source/speech/commands.py

```diff
@@ -180,13 +181,27 @@ def __eq__(self, __o: object) -> bool:
 class LangChangeCommand(SynthParamCommand):
 	"""A command to switch the language within speech."""
 
-	def __init__(self, lang: Optional[str]):
+	def __init__(self, lang: str | None):
 		"""
-		@param lang: the language to switch to: If None then the NVDA locale will be used.
+		:param lang: The language to switch to: If None then the NVDA locale will be used.
 		"""
 		self.lang = lang
 		self.isDefault = not lang
 
+	@staticmethod
+	def shouldSwitchVoice() -> bool:
+		"""Determines if the current synthesizer should switch to the voice corresponding to the language of the text been read."""
+		return config.conf["speech"]["autoLanguageSwitching"]
+
+	@staticmethod
+	def shouldGetLanguageForTextInfo() -> bool:
+		"""Determines if NVDA should get the language of the text been read."""
+		return (
+			config.conf["speech"]["autoLanguageSwitching"]
+			or config.conf["speech"]["reportLanguage"]
+			or config.conf["speech"]["reportNotSupportedLanguage"] != ReportNotSupportedLanguage.OFF.value
```

Comment:
If language switching is disabled, we don't need to check reportNotSupportedLanguage

```suggestion
```